include/diaspora.php

```diff
@@ -827,6 +854,37 @@ function ($match) use ($item){
 	}
 
 	/**
+	 * @brief Checks for relative /people/* links to match local contacts or
+	 * prepends the remote host taken from the author link
+	 *
+	 * @param string $body The item body to replace links from
+	 * @param string $author_link The author link for missing local contact fallback
+	 */
+	public function replace_people_guid($body, $author_link) {
+		$return = preg_replace_callback("&\[url=/people/([^\[\]]*)\](.*)\[\/url\]&Usi",
+			function ($match) use ($author_link){
+				// $match
+				// 0 => '[url=/people/0123456789abcdef]Foo Bar[/url]'
+				// 1 => '0123456789abcdef'
+				// 2 => 'Foo Bar'
+				$handle = self::url_from_fcontact($match[1]);
+
+				if ($handle) {
+					$return = '@[url='.$handle.']'.$match[2].'[/url]';
+				}else {
```

Comment:
I remember hesitating yesterday, I wasn't sure anymore, thanks for the reminder.